tests/Functional/TransactionTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Tests\Functional;
+
+use Doctrine\DBAL\Driver\PDO;
+use Doctrine\DBAL\Exception\ConnectionLost;
+use Doctrine\DBAL\Platforms\MySQLPlatform;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+
+use function sleep;
+
+use const E_ALL;
+use const E_WARNING;
+use const PHP_VERSION_ID;
+
+class TransactionTest extends FunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        if ($this->connection->getDatabasePlatform() instanceof MySQLPlatform) {
+            return;
+        }
+
+        self::markTestSkipped('Restricted to MySQL.');
+    }
+
+    public function testCommitFailure(): void
+    {
+        if (PHP_VERSION_ID < 70413 && $this->connection->getDriver() instanceof PDO\MySQL\Driver) {
+            self::markTestSkipped('See https://bugs.php.net/bug.php?id=66528.');
+        }
+
+        $this->connection->executeStatement('SET SESSION wait_timeout=1');
+
+        $this->connection->beginTransaction();
+
+        // during the sleep MySQL will close the connection
+        sleep(2);
+
+        // prevent the PHPUnit error handler from handling the "MySQL server has gone away" warning
+        $this->iniSet('error_reporting', (string) (E_ALL & ~E_WARNING));
+
+        $this->expectException(ConnectionLost::class);
+        $this->connection->commit();
```

Comment:
How hard would it be to also cover the `rollBack` scenario? Wouldn't it be a simple as calling `rollBack` here?

Reply:
Yea, I think so (assuming that the extensions report the disconnect). Let me update the test.

Reply:
Apparently, not 🚀 . We need to fix https://github.com/doctrine/dbal/issues/4718 first.